js/witness_calculator.js

```diff
@@ -119,15 +150,13 @@ class WitnessCalculator {
         this.i32 = new Uint32Array(memory.buffer);
         this.instance = instance;
 
-        this.n32 = (this.instance.exports.getFrLen() >> 2) - 2;
-        const pRawPrime = this.instance.exports.getPRawPrime();
-
+        this.n32 = this.instance.exports.getFieldNumLen32();
+        this.instance.exports.getRawPrime();
         const arr = new Array(this.n32);
         for (let i=0; i<this.n32; i++) {
-            arr[this.n32-1-i] = this.i32[(pRawPrime >> 2) + i];
+            arr[this.n32-1-i] = this.instance.exports.readSharedRWMemory(i);
         }
-
-        this.prime = Scalar.fromArray(arr, 0x100000000);
+        this.prime = fromArray32(arr);
```

Comment:
Why using the new `fromArray` instead of the Scalar.fromArray?

Reply:
They appear to be the same functionally, fixed! Switched both instances of new call to old one.